Branch-SDK/Branch-SDK/BNCNetworkInterface.m

```diff
@@ -0,0 +1,123 @@
+//
+//  BNCNetworkInterface.m
+//  Branch
+//
+//  Created by Ernest Cho on 11/19/19.
+//  Copyright © 2019 Branch, Inc. All rights reserved.
+//
+
+#import "BNCNetworkInterface.h"
+#import "BNCLog.h"
+
+#import <net/if.h>
+#import <ifaddrs.h>
+#import <arpa/inet.h>
+#import <netinet/in.h>
+
+typedef NS_ENUM(NSInteger, BNCNetworkAddressType) {
+    BNCNetworkAddressTypeUnknown = 0,
+    BNCNetworkAddressTypeIPv4,
+    BNCNetworkAddressTypeIPv6
+};
+
+@interface BNCNetworkInterface()
+
+@property (nonatomic, copy, readwrite) NSString *interfaceName;
+@property (nonatomic, assign, readwrite) BNCNetworkAddressType addressType;
+@property (nonatomic, copy, readwrite) NSString *address;
+
++ (NSArray<BNCNetworkInterface *> *)currentInterfaces;
+
+@end
+
+@implementation BNCNetworkInterface
+
+- (NSString*) description {
+    return [NSString stringWithFormat:@"<%@ %p %@ %@>",
+        NSStringFromClass(self.class),
+        self,
+        self.interfaceName,
+        self.address
+    ];
+}
+
++ (NSArray<BNCNetworkInterface *> *)currentInterfaces {
+
+    struct ifaddrs *interfaces = NULL;
+    NSMutableArray *currentInterfaces = [NSMutableArray arrayWithCapacity:8];
+
+    // Retrieve the current interfaces - returns 0 on success
+    if (getifaddrs(&interfaces) != 0) {
+        int e = errno;
+        BNCLogError(@"Can't read ip address: (%d): %s.", e, strerror(e));
+        goto exit;
+    }
+
+    // Loop through linked list of interfaces --
+    struct ifaddrs *interface = NULL;
+    for (interface=interfaces; interface; interface=interface->ifa_next) {
+        
+        // BNCLogDebugSDK(@"Found %s: %x.", interface->ifa_name, interface->ifa_flags);
```

Comment:
Forgot to uncomment/remove?

Reply:
Existing code and comments that I just moved wholesale.